planex/tarball.py

```diff
@@ -0,0 +1,96 @@
+"""
+tarball: Utilities for tar archives
+"""
+
+import fileinput
+import logging
+import os
+import sys
+import tarfile
+
+
+class Tarball(object):
+    """Represents a source archive tarball"""
+    def __init__(self, filename, prefix=""):
+        self.filename = filename
+        self.tarfile = tarfile.open(self.filename)
+        self.archive_root = archive_root(self.tarfile)
+        self.prefix = prefix
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        self.close()
+
+    def close(self):
+        """
+        Close the tarball
+        """
+        self.tarfile.close()
+
+    def extractfile(self, source):
+        """
+        Extract a file from the patchqueue, returning a file-like object
```

Comment:
I think we should make this module more general and say `tarball` (instead of `patchqueue`) here and in the next docstring